### PR TITLE
feat: adds Storybook

### PR DESCRIPTION
**Pull request by jkilzi**
https://issues.redhat.com/browse/FLPATH-701

This PR adds Storybook.
Two new scripts have been added to the root package.json:  
- `yarn start:storybook [plugins/<plugin-name-here>]`
- `yarn build:storybook [plugins/<plugin-name-here>]`

In order to see it in action, run `yarn start:storybook plugins/orchestrator`.
For more info on how to write stories:
- Check out: `plugins/orchestrator/src/components/OrchestratorPage/OrchestratorPage.stories.tsx`
- https://storybook.js.org/docs/react/writing-stories/introduction

image

### CODE REVIEW

File: packages/storybook/package.json
Comment (caponetto) [inline packages/storybook/package.json:28]:
```suggestion
    "@testing-library/jest-dom": "^5.17.0",
```
Other packages are using this version. Can we do the same? It would be easier to bump versions at once.

File: packages/storybook/package.json
Comment (caponetto) [inline packages/storybook/package.json:7]:
```suggestion
    "role": "common-library"
```
Maybe this role would suit this package better?

File: packages/storybook/package.json
Comment (caponetto) [inline packages/storybook/package.json:48]: I haven't seen any other package with custom prettier config. Do we need it?

File: packages/storybook/examples/components/Configure.mdx
Comment (caponetto) [inline packages/storybook/examples/components/Configure.mdx:1]: Do we need to add this `examples/` folder?

File: packages/storybook/package.json
Comment (jkilzi) [reply packages/storybook/package.json:7]: It is not a `common-library`. It is not meant to be consumed by other plugins or packages. It is actually an app.

File: packages/storybook/package.json
Comment (jkilzi) [reply packages/storybook/package.json:28]: IDK. I'll owe you an answer on this one, I'll check it and I'll get back to you.

File: packages/storybook/package.json
Comment (jkilzi) [reply packages/storybook/package.json:48]: We don't, I'll remove it

File: packages/storybook/package.json
Comment (jkilzi) [reply packages/storybook/package.json:7]: According to: https://backstage.io/docs/local-dev/cli-build-system#package-roles:  
>`common-library`: Isomorphic library for use by other packages

This package contains no isomorphic code. It's meant to be used entirely by frontend plugins as a mean to render components individually.

File: packages/storybook/package.json
Comment (jkilzi) [reply packages/storybook/package.json:28]: image

Long story short, what will get installed eventually is `5.17.0` (by the time of this message or until they release a new version).

Comment (jkilzi) [reply packages/storybook/package.json:28]: Requesting `^5.10.1` means that at least this package needs a version of `@testing-library/jest-dom` that is no older than 5.10.1.

File: plugins/orchestrator/package.json
Comment (jkilzi) [reply plugins/orchestrator/package.json:72]: Following the conversation [here about semver](https://github.com/caponetto/backstage-plugins/pull/24/files/8022783f1c7271a5bf30b0af95f2eeb8050bd56d#r1398332885), any package that is using `@backstage/test-utils:^1.4.4` will eventually be using `1.4.5` (assuming that `1.4.5` is the *latest in the version range*). This happens mostly due to how module hoisting works and especially inside a monorepo.

Comment (jkilzi) [reply plugins/orchestrator/package.json]: addressed in https://github.com/caponetto/backstage-plugins/pull/24/commits/f064fe4b27689f062d8d63ce370873ace0c26c6b

File: packages/storybook/package.json
Comment (jkilzi) [reply packages/storybook/package.json]: addressed in https://github.com/caponetto/backstage-plugins/pull/24/commits/f064fe4b27689f062d8d63ce370873ace0c26c6b

File: packages/storybook/package.json
Comment (jkilzi) [reply packages/storybook/package.json]: addressed in https://github.com/caponetto/backstage-plugins/pull/24/commits/f064fe4b27689f062d8d63ce370873ace0c26c6b

File: packages/storybook/examples/components/Configure.mdx
Comment (jkilzi) [reply packages/storybook/examples/components/Configure.mdx]: addressed in https://github.com/caponetto/backstage-plugins/pull/24/commits/f064fe4b27689f062d8d63ce370873ace0c26c6b

**Comment by caponetto** [comment]
Thanks @jkilzi, the code lgtm
There is only this issue with lint before we can merge

**Comment by jkilzi** [comment]
> Thanks @jkilzi, the code lgtm There is only this issue with lint before we can merge

10x, I'm working on it.

**Review by caponetto — APPROVED** [review]
🎉

**Comment by jkilzi** [comment]
I'm putting this on hold in order to favor https://github.com/janus-idp/backstage-plugins/pull/956

**Review by yzhao583 — APPROVED** [review]
LGTM thanks!

**Comment by mareklibra** [comment]
@jkilzi I think we can close this one in favor of https://github.com/janus-idp/backstage-plugins/pull/956, right?

**Pull request closed without merging**
